pkg/vault/test/foundry/unit/HooksConfigLibHooksHelpers.t.sol

```diff
@@ -0,0 +1,1072 @@
+            hookAdjustedAmountCalculatedRaw,
+            params,
+            state,
+            poolData
+        );
+    }
+
+    function _getParamsForCallAfterSwapHook(
+        SwapKind kind,
+        uint256 limitRaw
+    )
+        internal
+        pure
+        returns (
+            uint256 amountCalculatedScaled18,
+            uint256 amountCalculatedRaw,
+            uint256 hookAdjustedAmountCalculatedRaw,
+            SwapParams memory params,
+            SwapState memory state,
+            PoolData memory poolData
+        )
+    {
+        amountCalculatedScaled18 = 1e18;
+        amountCalculatedRaw = 2e18;
+        hookAdjustedAmountCalculatedRaw = 12e18;
+
+        params.kind = kind;
+        params.tokenIn = IERC20(address(0x01));
+        params.tokenOut = IERC20(address(0x02));
+        params.limitRaw = limitRaw;
+
+        state.indexIn = 0;
+        state.indexOut = 1;
+        state.amountGivenScaled18 = 1e18;
+
+        poolData.balancesLiveScaled18 = new uint256[](2);
+    }
+
+    function _callAfterSwapHook(
+        PoolConfigBits config,
+        uint256 amountCalculatedScaled18,
+        uint256 amountCalculatedRaw,
+        uint256 hookAdjustedAmountCalculatedRaw,
+        SwapParams memory params,
+        SwapState memory state,
+        PoolData memory poolData
+    ) internal returns (uint256 value) {
+        (uint256 amountInScaled18, uint256 amountOutScaled18) = params.kind == SwapKind.EXACT_IN
+            ? (state.amountGivenScaled18, amountCalculatedScaled18)
+            : (amountCalculatedScaled18, state.amountGivenScaled18);
+
+        vm.mockCall(
+            hooksContract,
+            abi.encodeWithSelector(
+                IHooks.onAfterSwap.selector,
+                AfterSwapParams({
+                    kind: params.kind,
+                    tokenIn: params.tokenIn,
+                    tokenOut: params.tokenOut,
+                    amountInScaled18: amountInScaled18,
+                    amountOutScaled18: amountOutScaled18,
+                    tokenInBalanceScaled18: poolData.balancesLiveScaled18[state.indexIn],
+                    tokenOutBalanceScaled18: poolData.balancesLiveScaled18[state.indexOut],
+                    amountCalculatedScaled18: amountCalculatedScaled18,
+                    amountCalculatedRaw: amountCalculatedRaw,
+                    router: router,
+                    pool: params.pool,
+                    userData: params.userData
+                })
+            ),
+            abi.encode(true, hookAdjustedAmountCalculatedRaw)
+        );
+
+        return
+            hooksConfigLibMock.callAfterSwapHook(
+                config,
+                amountCalculatedScaled18,
+                amountCalculatedRaw,
+                router,
+                params,
+                state,
+                poolData,
+                IHooks(hooksContract)
+            );
+    }
+
+    function _callAfterSwapHookAndExpectRevert(
```

Comment:
This one expects revert when applying limits.

There's one more case missing: the after swap hook returning false.